addons/popochiu/engine/interfaces/i_room.gd

```diff
@@ -198,6 +198,10 @@ func room_readied(room: PopochiuRoom) -> void:
 	if not is_instance_valid(current):
 		current = room
 	
+	# Apply a room script_name if it hasn't been explicitly set for the room
+	if room.script_name == "":
+		room.script_name = room.state.script_name
```

Comment:
Maybe we could check if `room.state.script_name` is not empty. In case it is, we could assume the `script_name` for the room as the name of its `.tscn` file (`room.scene_file_path`) without the `room_` preffix.

Reply:
Thinking about this further, I think the script name should be changed (I'm not sure how at this point) by the tool part of the code rather than at runtime. I have this sort of code in my scripts for setting the player's start position in a room:

```
if C.player.last_room == R.LockerRoom.script_name:
```
If in my room (i.e. "LockerRoom") the script name is not set, then I expect this test to fail. I haven't tested this yet though.